### PR TITLE
feat: add habit entry modification with notes and images

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR

Added ability to upload notes & pictures!!!

I think this is really good because it greatly improves the social aspect. For example, if you're in a habit working out with friends, now you can post a pic of you at the gym as proof and write a note saying what you did

Another important thing: I basically removed the ability to set a certain amount of completions per day. Instead of the square filling up the more completions you do in a day, it just shows a number with how many times you've completed today (see demo).

https://github.com/user-attachments/assets/6500e5d8-9370-4d98-8059-865523ea526f

Maybe we just make it a boolean option to allow multiple completions in a day, since for some habits it doesn't make sense to have multiple a day

This is better because different people doing a habit together might have slightly different goals... like if people want to track how many glasses of water they drink, maybe one person has a "goal" to drink 5 glasses a day, but for another person it's different

### What changed?

- Added support for adding notes and images to habit entries
- Enhanced habit completion tracking to store more detailed entry information
- Integrated expo-image-picker for capturing and selecting images
- Created a new modal interface for modifying habit entries
- Updated the habit card UI to accommodate the new entry modification features
- Restructured habit completion data model to support additional entry metadata

### How to test?

1. Open any habit card
2. Click the "Modify entry or add note/image" button
3. Try modifying the number of completions
4. Add a note to the entry
5. Take or select a photo for the entry
6. Navigate between different days using the arrow buttons
7. Save changes and verify they persist
8. Test camera permissions and image picker functionality

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/145?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#145** <a href="https://app.graphite.dev/github/pr/habits-together/app/145?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/145?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#144** <a href="https://app.graphite.dev/github/pr/habits-together/app/144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]
